wrappers/python/simtk/openmm/app/internal/customgbforces.py

```diff
@@ -342,12 +342,19 @@ def _mbondi3_radii(topology, all_bonds = None):
         all_bonds = _get_bonded_atom_list(topology)
     radii = _mbondi2_radii(topology, all_bonds=all_bonds)
     for i, atom in enumerate(topology.atoms()):
-        # carboxylate and HH/HE (ARG)
-        if _is_carboxylateO(atom, all_bonds):
-            radii[i] = 1.4
+        if atom.residue.name in ('GLU', 'ASP', 'GL4', 'AS4'):
+            if atom.name.startswith('OE') or atom.name.startswith('OD'):
+                radii[i] = 1.4
         elif atom.residue.name == 'ARG':
             if atom.name.startswith('HH') or atom.name.startswith('HE'):
                 radii[i] = 1.17
+        
+        # Adjust carboxylate O radii on C-Termini. Don't just do the end
+        # residue, since we can have C-termini in the middle as well
+        # (i.e., 2-chain dimers)
+        if atom.name == 'OXT':
```

Comment:
This line is dangerous because it just looks at the atom name without considering what residue it's in.  Atom names have no defined meaning except in the context of a particular residue.  I assume this is only meant to apply to proteins?  If so we should also check the residue name to make sure it's a protein residue.

Reply:
Yes, probably. I must confess I don't completely understand why the [i-1] radius gets changes as well.

Reply:
The `[i - 1]` radius gets changed because in the Amber files, the "other" terminal carboxylate oxygen comes immediately before the OXT atom name.

This solution is Amber-specific, and the original code *should* properly handle everything.  If it doesn't, then `_is_carboxylateO` should be fixed so that it identifies all carboxylate oxygens.